packages/components/src/palette-edit/test/index.tsx

```diff
@@ -82,18 +83,282 @@ describe( 'getNameForPosition', () => {
+				level: 5,
+				name: 'Test label',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'shows empty message', () => {
+		render(
+			<PaletteEdit
+				{ ...defaultProps }
+				emptyMessage={ 'Test empty message' }
+			/>
+		);
+
+		expect( screen.getByText( 'Test empty message' ) ).toBeInTheDocument();
+	} );
+
+	it( 'can remove all colors', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Remove all colors',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'can reset colors', async () => {
+		const user = userEvent.setup();
+		render(
+			<PaletteEdit { ...defaultProps } colors={ colors } canReset />
+		);
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Reset colors',
+			} )
+		).toBeInTheDocument();
+	} );
+
+	it( 'can not reset colors', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Reset colors',
+			} )
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'can add a new color palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Add color',
+			} )
+		);
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				...colors,
+				{
+					color: '#000',
+					name: 'Color 1',
+					slug: 'color-1',
+				},
+			] );
+		} );
+	} );
+
+	it( 'can add a new graident palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } gradients={ gradients } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Add gradient',
+			} )
+		);
+
+		await waitFor( () => {
+			expect( defaultProps.onChange ).toHaveBeenCalledWith( [
+				...colors,
+				{
+					color: '#000',
+					name: 'Color 1',
+					slug: 'color-1',
+				},
+			] );
+		} );
+	} );
+
+	it( 'can not add a new palette', () => {
+		render( <PaletteEdit { ...defaultProps } canOnlyChangeValues /> );
+
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Add color',
+			} )
+		).not.toBeInTheDocument();
+	} );
+
+	it( 'can delete palette', async () => {
+		const user = userEvent.setup();
+		render( <PaletteEdit { ...defaultProps } colors={ colors } /> );
+
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Color options',
+			} )
+		);
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Show details',
+			} )
+		);
+		await user.click( screen.getByText( 'Primary' ) );
```

Comment:
This is a great example of our tests doubling as accessibility tests. The fact that there's no labeled button to click means that this part of the UI is not accessible. Let's fix this in a follow-up.